projects/js-packages/publicize-components/src/components/form/index.tsx

```diff
@@ -25,6 +27,25 @@ import { SharePostForm } from './share-post-form';
 import styles from './styles.module.scss';
 import { UnsupportedConnectionsNotice } from './unsupported-connections-notice';
 
+const findPostMedia = content => {
+	const parser = new DOMParser();
+	const doc = parser.parseFromString( content, 'text/html' );
+	const imgElements = Array.from( doc.querySelectorAll( 'img' ) );
+
+	if ( imgElements.length === 0 ) {
+		return null;
+	}
+
+	const imgElement = imgElements[ 0 ];
+	const classNames = imgElement.className.split( ' ' );
+	const imageClass = classNames.find( className => className.startsWith( 'wp-image-' ) );
+	if ( imageClass ) {
+		const mediaId = parseInt( imageClass.replace( 'wp-image-', '' ) );
+		const mediaUrl = imgElement?.src;
+		return { mediaId, mediaUrl };
+	}
+};
+
```

Comment:
I'm not sure if there is better way of doing this, for the POC I just got the image from the post content